### ISSUE TITLE
Other operating systems?

### ISSUE THREAD

**Issue by da2x** [TechnitiumSoftware/Mesh#5]
Any plans to support other operating systems? People use different operating systems and it’s impossible to ask a Linux user to switch to Windows just for a chat app.

**Comment by ShreyasZare** [comment]
Yes, there is a plan to have a cross platform app for platforms that can run .NET Core. Mesh's predecessor, Bit Chat, had a Linux app that could run on Mono Framework. But, Mono's implementation for Windows Forms is such a mess that it was not used for Mesh.

Since, .NET Core does not support cross platform GUI forms yet, Mesh is planned to have web based UI that will access locally running Mesh web server. This way, Mesh could be used on any platform.

Timeline for this is not clear yet since I am currently working on the DNS Server project.

**Comment by ASHIT-AXAR** [comment]
@ShreyasZare i'm suggesting the Vala language. very similar to C#

https://wiki.gnome.org/Projects/Vala/ValaForCSharpProgrammers

**Comment by ShreyasZare** [comment]
> @ShreyasZare i'm suggesting the Vala language. very similar to C#
> 
> https://wiki.gnome.org/Projects/Vala/ValaForCSharpProgrammers

Well, its never just about the language. With .NET there is its standard library and ecosystem which wont be available otherwise. Plus there is years of code written which will need rewrite which makes it unfeasible.
